### PR TITLE
Replace twig component header

### LINKED ISSUE #33103: Twig Component: Header (from `<head></head>`)
## Description

See #33101 for description

This is the head part (from the `<head></head>` content).

### PR DESCRIPTION
**Pull request by boherm**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Replace header by twig component header
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI ✅ & Autotests ✅
| Fixed ticket?     | Fixes #33103 
| Related PRs       | 
| Sponsor company   | PrestaShop SA

### CODE REVIEW

**Review by M0rgan01 — APPROVED** [review]

File: classes/controller/Controller.php [hunk 1]
```diff
@@ -644,7 +644,7 @@ protected function smartyOutputContent($templates)
         $this->context->cookie->write();
 
         $js_tag = 'js_def';
-        $this->context->smarty->assign($js_tag, $js_tag);
+        $this->context->smarty->assign($js_tag, \Media::getJsDef());
```
Comment (lartist) [inline classes/controller/Controller.php:647]: \Media::getJsDef()) return an empty array instead of 'js_def'. Is it normal ?

**Review by matthieu-rolland — COMMENTED** [review]
lgtm, minus @lartist 's question

File: classes/controller/Controller.php (hunk 1 above)
Comment (boherm) [reply classes/controller/Controller.php:647]: I have found that `js_def` must be an array and can not be a string. (And have a variable `js_def` with the value `"js_def"` is dumb...)
This feature aim to define some JS variables directly on layout head. Also, i'm pretty sure that isn't be very used, so maybe we can simplify.

Comment (lartist) [reply classes/controller/Controller.php:647]: Ok @boherm , it's ok for me to try correcting a related bug in this PR.
Maybe we should notice QA there is special check on this PR.

**Review by lartist — APPROVED** [review]

**Comment by hibatallahAouadni** [comment]
Hello @boherm & @lartist 

The auto-tests are failing everywhere: https://github.com/hibatallahAouadni/testing_pr/actions/runs/5722851413
Please check and feedback.

Thanks!

**Comment by boherm** [comment]
Tests legacy: https://github.com/boherm/ga.tests.ui.pr/actions/runs/5728919470
Tests symfony: https://github.com/boherm/ga.tests.ui.pr/actions/runs/5729425583

**Comment by hibatallahAouadni** [comment]
@boherm the legacy tests are all :green_circle: 
But the symfony :disappointed: the majority are failing :red_circle: 
@PrestaShop/qa-automation is it related to the PR?

**Comment by Progi1984** [comment]
@hibatallahAouadni Nope. `Symfony` is not stable, it's why it's behind a feature flag. 

We agreed with @jolelievre to let them :red_circle: for the moment.

**Comment by hibatallahAouadni** [comment]
Thanks @Progi1984 for your quick answer 🚀 
Well then, CI and auto tests are 🟢 so it's a GO from QA 🎉

**Comment by nicosomb** [comment]
Thank you @boherm

**Pull request merged**